### ISSUE TITLE
category value in column name?

### ISSUE THREAD

**Issue by kodonnell** [scikit-learn-contrib/categorical-encoding#42]
E.g.

```
>>> df = pd.DataFrame({'col': ['a', 'b']})
>>> encoder = ce.OneHotEncoder()
>>> encoder.fit_transform(df)
   col_0  col_1  col_-1
0      1      0       0
1      0      1       0
```

Ignoring last column, I'd like to see

```
   col_a  col_b
0      1      0
1      0      1
```

So that e.g. if I find `col_a` is a strong feature in a model, I can interpret what it means easily (as opposed to `col_0`). I know I can hack this myself, but I'd have thought it'd be a common use case, and natively supported. Unless I'm missing something ...

**Comment by wdm0006** [comment]
This has come up before, notably here: https://github.com/scikit-learn-contrib/categorical-encoding/issues/21

Then it was implemented in the two encoders where I could figure out how to here: https://github.com/scikit-learn-contrib/categorical-encoding/commit/056a483eff86f1ca1f57d7b7c0fab520fa988200

For something like binary encoding the output columns don't map 1-to-1 with categories, so it doesn't quite work out. 

Does that help?

**Comment by kodonnell** [comment]
Sorry, I missed that issue when looking.

I'd still like to have an option to use variable values instead of indices, and leave the responsibility to the user. (Though any string value is supported as a name in pandas, correct?) In my case, they're nearly always 'nice'. This is the standard behaviour of other things I've used (e.g. R or `pandas.get_dummies`). Anyway, because this isn't supported, I'm rolling my own mixin (in part as I need to roll a custom one anyway for other things).

Happy to close this issue if you disagree.

**Comment by wdm0006** [comment]
No worries, if left as an option that defaults to current behavior I’d not be opposed to including that kind of thing for the encoders where it makes sense, if you’re interested in working on a PR

**Comment by kodonnell** [comment]
won't be able to offer a PR sorry (at least in next few months)

**Comment by wdm0006** [comment]
Ok, I can work on one if I get a chance, but am marking this as seeking a contributor in the meantime. If anyone would like to work on this, chime in here.

**Comment by wdm0006** [comment]
Merged PR #65
